### ISSUE TITLE
Error while doing a few-shot training with the new tars-model

### ISSUE THREAD

**Issue by manojlds** [flairNLP/flair#2039]
**Describe the bug**
While trying to do few-shot training as mentioned in the [TARS tutorial](https://github.com/flairNLP/flair/blob/v0.7/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md) getting the following error:

```
KeyError                                  Traceback (most recent call last)
<ipython-input-33-5a476c50848a> in <module>()
      9               mini_batch_size=1,
     10               max_epochs=10,
---> 11               train_with_dev=True,
     12               )

3 frames
/usr/local/lib/python3.6/dist-packages/flair/models/text_classification_model.py in _get_nearest_labels_for(self, labels)
    684             plausible_labels = []
    685             plausible_label_probabilities = []
--> 686             for plausible_label in self.label_nearest_map[label]:
    687                 if plausible_label in already_sampled_negative_labels:
    688                     continue

KeyError: 'Sci/Tech'
```

**To Reproduce**
https://colab.research.google.com/drive/1_YuHMKhF1JajAU0KBp3bsM7qpqnfhKYu

**Expected behavior**
Should train the model

**Environment (please complete the following information):**
Colab

**Comment by kishaloyhalder** [comment]
Hi @manojlds 

Thanks for trying out the model and reaching out about the issue!

I checked out the colab notebook and found the core issue. The tutorial shows you how to train the TARS model on a new task (called `FOOD_DRINK`) which does not already exist in the pre-trained model. However, your code is trying to add a task called `NEWS_CATEGORY` which already exists in the model.

```python
In [3]: tars.list_existing_tasks()                                                                                                              
Out[3]: 
{'AGNews',
 'Amazon',
 'DBPedia',
 'GO_EMOTIONS',
 'IMDB',
 'NEWS_CATEGORY',
 'SST',
 'TREC_6',
 'Yelp'}
```
By design, when it finds a task with the same name in the memory, it just switches to it with a notification.
as logged in your output: ```Task `NEWS_CATEGORY` already exists in TARS model. Switching to it.```
This task has a completely different set of labels as the following:

```python
In [8]: print(tars.task_specific_attributes['NEWS_CATEGORY']['label_dictionary'])                                                               
Dictionary with 41 tags: ENTERTAINMENT, PARENTING, MONEY, POLITICS, WOMEN, TRAVEL, COMEDY, WORLDPOST, HEALTHY_LIVING, QUEER_VOICES, BUSINESS, WORLD_NEWS, WEIRD_NEWS, WEDDINGS, PARENTS, BLACK_VOICES, FOOD_&_DRINK, WELLNESS, RELIGION, CRIME, CULTURE_&_ARTS, HOME_&_LIVING, IMPACT, SPORTS, STYLE, THE_WORLDPOST, STYLE_&_BEAUTY, GOOD_NEWS, DIVORCE, MEDIA
```

I think you meant the `AGNews` corpus which is also part of the TARS model with a slightly different set of labels.

```python
In [9]: print(tars.task_specific_attributes['AGNews']['label_dictionary'])                                                                      
Dictionary with 4 tags: Sports, Business, Science_Technology, World
```
So there are two options that I can recommend:

1. Rename the label 'Sci/Tech' to 'Science_Technology' in your corpus, and just switch to the 'AGNews' task before starting to train. We did this during training as TARS tries to infer the semantics of the class from the label name ([reference](https://www.aclweb.org/anthology/2020.coling-main.285.pdf)).
2. Add a new task as per your definition. So this statement  ```tars.add_and_switch_to_new_task("NEWS_CATEGORY", label_dictionary=corpus.make_label_dictionary())``` should become ```tars.add_and_switch_to_new_task("MY_NEWS_CATEGORY", label_dictionary=corpus.make_label_dictionary())```. 

Hope this helps. Let me know if you need anything else.

with regards,
Kishaloy

**Comment by manojlds** [comment]
Ah, thanks. I did notice the `Task `NEWS_CATEGORY` already exists in TARS model. Switching to it.` message but thought it must have been due to a previous run of the cell since I was experimenting with hyperparams a bit. Thanks for the context, didn't think that a name I had given randomly would already exist :)

Got it working with both the options 👍
